compiler/AST/baseAST.cpp

```diff
@@ -570,6 +570,13 @@ void update_symbols(BaseAST* ast, SymbolMap* map) {
   if (SymExpr* sym_expr = toSymExpr(ast)) {
     if (sym_expr->symbol()) {
       if (Symbol* y = map->get(sym_expr->symbol())) {
+        // If the symbol is re-maped again (e.g., x was y and y was z),
+        // we need to keep looking until we find the final symbol.
+        while (Symbol* z = map->get(y)) {
+          if (z == y) break;
+          y = z;
+        }
+
```

Comment:
This introduces some asymmetry because this is not the only place where a replacement occurs in this function. See LoopStmt and ForallStmt branches below.

I like this change, but I think it should be consistent across branches.